### PR TITLE
DataProfiler: `graph_data_demo` update

### PR DESCRIPTION
**Pull request by micdavis**
- Updated the `graph_data_demo.py` to show off the save, load, and diff functionality added to the graph profiler

### CODE REVIEW

File: examples/graph_data_demo.ipynb [hunk 1]
```diff
@@ -24,31 +32,46 @@
...
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
```
Comment (JGSweets) [inline examples/graph_data_demo.ipynb:62]: what's the pycharm stuff?

File: examples/graph_data_demo.ipynb [hunk 2]
```diff
@@ -139,17 +174,168 @@
...
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Diff\n",
```
Comment (JGSweets) [inline examples/graph_data_demo.ipynb:223]: let's follow the pattern of other notebooks and make these more like phrases:
`Differences in Data`

File: examples/graph_data_demo.ipynb [hunk 3]
```diff
@@ -139,17 +174,168 @@
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "source": [
+    "## Save and Load\n",
```
Comment (JGSweets) [inline examples/graph_data_demo.ipynb:178]: let's follow the pattern of other notebooks and make these more like phrases:
`Saving and Loading a Profile`

File: examples/graph_data_demo.ipynb [hunk 4]
```diff
@@ -139,17 +174,168 @@
...
+   }
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
     "We have shown the graph pipeline in the DataProfiler. It works similarly to the current DataProfiler implementation."
-   ]
+   ],
+   "metadata": {
```
Comment (JGSweets) [inline examples/graph_data_demo.ipynb:321]: remove extra cell

File: examples/graph_data_demo.ipynb [hunk 5]
```diff
@@ -139,17 +174,168 @@
...
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Diff\n",
+    "If we wanted to ensure that this new profile was the same as the previous profile that we loaded, we could compare them using the diff functionality."
```
Comment (JGSweets) [inline examples/graph_data_demo.ipynb:224]: maybe can update this to have more of a story like:

If we

>  wanted to ensure that this new profile was the same as the previous profile that we loaded, we could compare them using the diff functionality.
> 
> Such reports can provide details on the differences between training and validation data like in this pseudo example:
> ```
> profiler_training = dp.Profiler(training_data)
> profiler_testing = dp.Profiler(testing_data)
> 
> validation_report = profiler_training.diff(profiler_testing)
> ```

File: examples/graph_data_demo.ipynb (hunk 1 above)
Comment (micdavis) [reply examples/graph_data_demo.ipynb:62]: I removed it in vim, but when I committed it all got added back in. I think it is added in by the pre-commits

File: examples/graph_data_demo.ipynb (hunk 5 above)
Comment (micdavis) [reply examples/graph_data_demo.ipynb:430]: Added in some context to make it more meaningful

File: examples/graph_data_demo.ipynb [hunk 6]
```diff
@@ -139,17 +150,124 @@
...
    "cell_type": "markdown",
-   "metadata": {},
+   "source": [
+    "## Saving and Loading a Profile\n",
+    "Below you will see an example of how a Graph Profile can be saved and loaded again."
+   ],
+   "metadata": {
+    }
+   }
+  },
```
Comment (taylorfturner) [inline examples/graph_data_demo.ipynb:160]: this looks odd....

**Review by taylorfturner — APPROVED** [review]

**Review by ksneab7 — APPROVED** [review]

**Pull request merged**